crates/node-core/src/init.rs

```diff
@@ -96,16 +98,16 @@ pub fn init_genesis<DB: Database>(factory: ProviderFactory<DB>) -> Result<B256,
 }
 
 /// Inserts the genesis state into the database.
-pub fn insert_genesis_state<DB: Database>(
+pub fn insert_genesis_state<'a, 'b, DB: Database>(
     tx: &<DB as Database>::TXMut,
-    genesis: &reth_primitives::Genesis,
+    capacity: usize,
```

Comment:
I don't think we need this, we could use `let capacity = alloc.size_hint().and_then(|(_,hint)|hint).unwrap_or_default();`